"use strict";var e=require("@ethersproject/logger"),r=require("@privy-io/js-sdk-core"),s=require("../errors.js");require("ofetch");class t extends s.PrivyConnectorError{constructor(){super("Wallet timeout"),this.type="wallet_error"}}class o extends s.PrivyConnectorError{constructor(){super("User rejected connection"),this.type="wallet_error"}}class i extends s.PrivyError{constructor(e,r,s){super(e),this.type="provider_error",this.code=r,this.data=s}}class n extends i{constructor(e){super(e.message,e.code,e.data);let s=Object.values(r.ProviderErrors).find((r=>r.eipCode===e.code));this.details=s||r.ProviderErrors.UNKNOWN_ERROR,-32002===e.code&&(e.message?.includes("already pending for origin")?e.message?.includes("wallet_requestPermissions")?this.details=r.ProviderErrors.E32002_CONNECTION_ALREADY_PENDING:this.details=r.ProviderErrors.E32002_REQUEST_ALREADY_PENDING:e.message?.includes("Already processing")&&e.message.includes("eth_requestAccounts")&&(this.details=r.ProviderErrors.E32002_WALLET_LOCKED))}}const a={ERROR_USER_EXISTS:{message:"User already exists for this address",detail:"Try another address!",retryable:!1},ERROR_TIMED_OUT:{message:"Wallet request timed out",detail:"Please try connecting again.",retryable:!0},ERROR_WALLET_CONNECTION:{message:"Could not log in with wallet",detail:"Please try connecting again.",retryable:!0},ERROR_USER_REJECTED_CONNECTION:{message:"You rejected the request",detail:"Please try connecting again.",retryable:!0},ERROR_USER_LIMIT_REACHED:{message:"Unable to link",detail:"You've reached the maximum number of linked wallets.",retryable:!1},...r.ProviderErrors};exports.ConnectorErrors=a,exports.PrivyProviderRpcError=n,exports.ProviderRpcError=i,exports.UserRejectedConnectionError=o,exports.WalletTimeoutError=t,exports.formatConnectorError=t=>{if(t instanceof s.PrivyConnectorError)return t;if(t?.code&&t?.reason){let s=new n(t);return t.code===e.ErrorCode.ACTION_REJECTED&&(s.details=r.ProviderErrors.E4001_USER_REJECTED_REQUEST),s}return t?.code?new n(t):new s.PrivyConnectorError("Unknown connector error",t)};
